fastly/custom_tls_configuration.go

```diff
@@ -0,0 +1,153 @@
+type ListCustomTLSConfigurationsInput struct {
+	FilterBulk *bool   // Whether or not to only include bulk=true configurations
+	Include    *string // Include related objects. Optional, comma-separated values. Permitted values: dns_records.
+	PageNumber *uint   // The page index for pagination.
+	PageSize   *uint   // The number of keys per page.
+
+}
+
+// formatFilters converts user input into query parameters for filtering.
+func (i *ListCustomTLSConfigurationsInput) formatFilters() map[string]string {
+	result := map[string]string{}
+	pairings := map[string]interface{}{
+		"filter[bulk]": i.FilterBulk,
+		"include":      i.Include,
+		"page[size]":   i.PageSize,
+		"page[number]": i.PageNumber,
+	}
+	for key, value := range pairings {
+		if !reflect.ValueOf(value).IsNil() {
+			result[key] = fmt.Sprintf("%v", reflect.ValueOf(value).Elem())
+		}
+	}
+	return result
+}
+
+// ListCustomTLSConfigurations list all TLS configurations.
+func (c *Client) ListCustomTLSConfigurations(i *ListCustomTLSConfigurationsInput) ([]*CustomTLSConfiguration, error) {
+	p := "/tls/configurations"
+	filters := &RequestOptions{
+		Params: i.formatFilters(),
+		Headers: map[string]string{
+			"Accept": "application/vnd.api+json", // this is required otherwise the filters don't work
+		},
+	}
+
+	r, err := c.Get(p, filters)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := jsonapi.UnmarshalManyPayload(r.Body, reflect.TypeOf(new(CustomTLSConfiguration)))
+	if err != nil {
+		return nil, err
+	}
+
+	con := make([]*CustomTLSConfiguration, len(data))
+	for i := range data {
+		typed, ok := data[i].(*CustomTLSConfiguration)
+		if !ok {
+			return nil, fmt.Errorf("unexpected response type: %T", data[i])
+		}
+		con[i] = typed
+	}
+
+	return con, nil
+}
+
+// GetCustomTLSConfigurationInput is used as input to the GetCustomTLSConfiguration function.
+type GetCustomTLSConfigurationInput struct {
+	ID      string
+	Include *string // Include related objects. Optional, comma-separated values. Permitted values: dns_records.
+}
+
+// GetCustomTLSConfiguration returns a single TLS configuration.
+func (c *Client) GetCustomTLSConfiguration(i *GetCustomTLSConfigurationInput) (*CustomTLSConfiguration, error) {
+	if i.ID == "" {
+		return nil, ErrMissingID
+	}
+
+	p := fmt.Sprintf("/tls/configurations/%s", i.ID)
+
+	ro := &RequestOptions{
+		Headers: map[string]string{
+			"Accept": "application/vnd.api+json", // this is required otherwise the params don't work
+		},
+	}
+
+	if i.Include != nil {
+		ro.Params = map[string]string{"include": *i.Include}
+	}
+
+	r, err := c.Get(p, ro)
+	if err != nil {
+		return nil, err
+	}
+
+	var con CustomTLSConfiguration
+	if err := jsonapi.UnmarshalPayload(r.Body, &con); err != nil {
+		return nil, err
+	}
+
+	return &con, nil
+}
+
+// UpdateCustomTLSConfigurationInput is used as input to the UpdateCustomTLSConfiguration function.
+type UpdateCustomTLSConfigurationInput struct {
+	ID   string `jsonapi:"attr,id"`
```

Comment:
Does `ID` require the jsonapi struct tag or can it be omitted? as the API docs for this endpoint suggests only `name` is a required attribute.

Reply:
You're right. The ID is not marshaled into the body, it just used in the path. Removed.

Reply:
Looks like the struct tag is still showing, maybe you need to re-push?